### PR TITLE
Proposal-1: Partitioner Refactoring

### PR DESCRIPTION
**Pull request by RobertIndie**
### Motivation

Please see the proposal in this PR

### Documentation

Check the box below.

Need to update docs?

- [ ] `doc-required`

  (If you need help on updating docs, create a doc issue)

- [ ] `no-need-doc`

  (Please explain why)

- [x] `doc`

  (If this PR contains doc changes)

### CODE REVIEW

File: proposals/Proposal-1-Time-Partitioner-Refactoring.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+To ensure backward compatibility, a `Legacy Partitioner` will be introduced. If users wish to revert to previous
+behavior, they can set the `partitioner` configuration to `legacy` to utilize the `Legacy Partitioner`. In this case,
+the `partitionerType` is respected, allowing users to revert to legacy behavior.
+
+This design's advantage is that it maintains the current partitioner implementation and makes it easier to remove the
+legacy partitioner implementation and `partitionType` configuration in the future.
+
+#### Configuration
+
+Add a new configuration `partitioner` to the Cloud Storage sink connector. The value of this configuration can
```
Comment (shibd) [inline proposals/Proposal-1-Time-Partitioner-Refactoring.md:136]: Good idea.

File: proposals/Proposal-1-Time-Partitioner-Refactoring.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+## High Level Design
+
+A new `Partitioner` interface will be added, with two partitioner implementations: `TopicPartitioner`
+and `TimePartitioner`.
+
+The behavior of these partitioners is as follows:
+
+- **Topic Partitioner**: Messages are partitioned according to the pre-existing partitions in the Pulsar topics. For
+  instance, a message for the topic `public/default/my-topic-partition-0` would be directed to the
+  file `public/default/my-topic-partition-0/xxx.json`, where `xxx` signifies the message offset.
```
Comment (shibd) [inline proposals/Proposal-1-Time-Partitioner-Refactoring.md:72]: > where `xxx` signifies the message offset.

Is this the offset that represents the earliest or the last offset?

**Review by shibd — DISMISSED** [review]
LGTM, A small question.

File: proposals/Proposal-1-Time-Partitioner-Refactoring.md (hunk 2 above)
Comment (RobertIndie) [reply proposals/Proposal-1-Time-Partitioner-Refactoring.md]: The earliest. I have also updated the proposal to make it clearer.

**Review by shibd — APPROVED** [review]

**Pull request merged**